### PR TITLE
Rendering: Add configuration options for `renderKey` lifetime

### LINKED ISSUE #57303: Panel Rendering failing due to expired RenderKey
**What happened**:

Panel rendering for long and complex queries is failing due to 401 errors during rendering. 

Context: 
I have a custom data source plugin that is making multiple requests to display the desired data. 
On the backend, the process is quite long as computing the series is not trivial (multiple source of data, lots of series, high resolution, ...). Each request the backend is doing takes 30s to 1 minute, but it requires querying many times the backend with different parameters (10 to 15 times). In the end, sometimes it takes 10 to 15 minutes to render the panel.

That being said, the panel is rendered as PNG using the image render feature.

In summary:
It fails with multiple 401 answers from the datasource proxy, meaning that the provided renderKey has expired.
This is confirmed with the following log line in Grafana server: `Failed to get the render key from cache`

**What you expected to happen**:

Render the panel successfully. 

**How to reproduce it (as minimally and precisely as possible)**:

In a nutshell:
1. Create an empty datasource plugin
2. Make multiple calls, with delay between (so that it take more than 5 minutes) them, to any API with the datasource proxy
3. Ask for panel to be rendered by Grafana image 

**Anything else we need to know?**:

I might have found a hard-coded limit for the life duration of the renderKey used when a request to the renderer is made.
It might need to become something configurable with current hard-coded value as default. 
https://github.com/grafana/grafana/blob/23956557d8c6a119b7de5be5c42024e29634d002/pkg/services/rendering/rendering.go#L84

**Environment**:
- Grafana version: 8 and 9 
- Data source type & version: Custom
- OS Grafana is installed on: Docker
- User OS & Browser: Firefox > 100
- Grafana plugins: Custom datasource
- Others: Image renderer 3.5.0

### PR DESCRIPTION
**Pull request by Willena**
**What this PR does / why we need it**: 

It fixes the mentioned issue, bu adding a new configuration key to specify the lifetime of the generated renderKey, so that it does not expire before the end of the panel/dashboard rendering process.

**Which issue(s) this PR fixes**: #57303 

Fixes #57303

### CODE REVIEW

**Review by joanlopez — APPROVED** [review]
Looks good as far as I can tell, thanks @Willena!

Just wondering if the setting should either be `render_key_lifetime`.

**Review by spinillos — APPROVED** [review]

**Comment by Willena** [comment]
@joanlopez would you like me to change the setting name ?

**Comment by joanlopez** [comment]
> @joanlopez would you like me to change the setting name ?

No one has raised any additional thought, so from my side I'd say: yes, please! 🙏🏻 

Thanks!!!

**Comment by Willena** [comment]
@joanlopez done !

File: conf/defaults.ini [hunk 1]
```diff
@@ -1123,6 +1123,10 @@ renderer_token = -
 # Concurrent render request limit affects when the /render HTTP endpoint is used. Rendering many images at the same time can overload the server,
 # which this setting can help protect against by only allowing a certain amount of concurrent requests.
 concurrent_render_request_limit = 30
+# RenderKey max duration expressed in duration (`10s`, `1m`)
+# The renderkey is used by the image renderer to access grafana (it's like an API key). The default validity is 5 minutes. This should be more than enough for most deployments.
+# Change the value only if image rendering is failing and you see `Failed to get the render key from cache` in grafana logs.
```
Comment (joanlopez) [inline conf/defaults.ini:1128]:
```suggestion
# Determines the life time of the render key used by the image renderer to access and render Grafana.
# This setting should be expressed as a duration. Examples: 10s (seconds), 5m (minutes), 2h (hours). 
# Default is 5m. This should be more than enough for most deployments.
# Change the value only if image rendering is failing and you see `Failed to get the render key from cache` in Grafana logs.
```

Comment (joanlopez) [reply conf/defaults.ini:1128]: Hey @Willena,

I suggested some minor changes on the setting's description.
Could you review if those make sense for you?

If so, could you update the `conf/sample.ini` accordingly, please?
You'll see that file also contains all the available settings with the default value.

Thanks!

**Comment by joanlopez** [comment]
> @joanlopez done !

Greatt! 👌🏻 

I think it looks really nice and we're quite close to merge it.

Could you please review my last few comments? 
Once we get those resolved I'll be fine with merging it.

Thanks!

File: conf/sample.ini [hunk 2]
```diff
@@ -1089,6 +1089,10 @@
 # Concurrent render request limit affects when the /render HTTP endpoint is used. Rendering many images at the same time can overload the server,
 # which this setting can help protect against by only allowing a certain amount of concurrent requests.
 ;concurrent_render_request_limit = 30
+# RenderKey max duration expressed in duration (`10s`, `1m`)
+# The renderkey is used by the image renderer to access grafana (it's like an API key). The default validity is 5 minutes. This should be more than enough for most deployments.
+# Change the value only if image rendering is failing and you see `Failed to get the render key from cache` in grafana logs.
```
Comment (joanlopez) [inline conf/sample.ini:1094]: This should match the description on `defaults.ini`, please 🙏🏻

Comment (Willena) [reply conf/sample.ini:1094]: Of course ! I've gone too quickly and I don't know how I missed that ! Sorry

**Pull request merged**